packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -0,0 +1,600 @@
+
+  void _handleTapDown(TapDownDetails details) {
+    _renderEditable.handleTapDown(details);
+  }
+
+  void _handleTap() {
+    _renderEditable.handleTap();
+    _requestKeyboard();
+  }
+
+  void _handleTapCancel() {
+  }
+
+  void _handleLongPress() {
+    _renderEditable.handleLongPress();
+  }
+
+  @override
+  bool get wantKeepAlive => _controller?.text?.isNotEmpty == true;
+
+  @override
+  void deactivate() {
+    super.deactivate();
+  }
+
+  bool _shouldAppearForModeAndText(OverlayVisibilityMode mode, bool hasText) {
+    switch (mode) {
+      case OverlayVisibilityMode.never:
+        return false;
+      case OverlayVisibilityMode.always:
+        return true;
+      case OverlayVisibilityMode.editing:
+        return hasText;
+      case OverlayVisibilityMode.notEditing:
+        return !hasText;
+    }
+    assert(false);
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context); // See AutomaticKeepAliveClientMixin.
+    assert(debugCheckHasDirectionality(context));
+    final Brightness keyboardAppearance = widget.keyboardAppearance;
+    final TextEditingController controller = _effectiveController;
+    final FocusNode focusNode = _effectiveFocusNode;
+    final List<TextInputFormatter> formatters = widget.inputFormatters ?? <TextInputFormatter>[];
+    final bool enabled = widget.enabled ?? true;
+    if (widget.maxLength != null && widget.maxLengthEnforced)
+      formatters.add(LengthLimitingTextInputFormatter(widget.maxLength));
+
+    Widget child = Padding(
+      padding: widget.padding,
+      child: RepaintBoundary(
+        child: EditableText(
+          key: _editableTextKey,
+          controller: controller,
+          focusNode: focusNode,
+          keyboardType: widget.keyboardType,
+          textInputAction: widget.textInputAction,
+          textCapitalization: widget.textCapitalization,
+          style: widget.style,
+          textAlign: widget.textAlign,
+          autofocus: widget.autofocus,
+          obscureText: widget.obscureText,
+          autocorrect: widget.autocorrect,
+          maxLines: widget.maxLines,
+          selectionColor: _kSelectionHighlightColor,
+          selectionControls: cupertinoTextSelectionControls,
+          onChanged: widget.onChanged,
+          onEditingComplete: widget.onEditingComplete,
+          onSubmitted: widget.onSubmitted,
+          inputFormatters: formatters,
+          rendererIgnoresPointer: true,
+          cursorWidth: widget.cursorWidth,
+          cursorRadius: widget.cursorRadius,
+          cursorColor: widget.cursorColor,
+          scrollPadding: widget.scrollPadding,
+          keyboardAppearance: keyboardAppearance,
+        ),
+      ),
+    );
+
+    if (widget.placeholder != null ||
+        widget.clearButtonMode != OverlayVisibilityMode.never ||
+        widget.prefix != null ||
+        widget.suffix != null) {
+      child = ValueListenableBuilder<TextEditingValue>(
```

Comment:
I think re-assignment of 'child' is probably unnecessarily confusing for readers.  Can we stick to immutable values as we compose the tree?

Reply:
Refactored